src/clients/BundleDataClient.ts

```diff
@@ -720,9 +598,10 @@ export class BundleDataClient {
             // older deposit in case the spoke pool client's lookback isn't old enough to find the matching deposit.
             if (fill.blockNumber >= destinationChainBlockRange[0]) {
               const historicalDeposit = await queryHistoricalDepositForFill(originClient, fill);
-              if (!historicalDeposit.found || !utils.isV3Deposit(historicalDeposit.deposit)) {
+              if (!historicalDeposit.found) {
                 bundleInvalidFillsV3.push(fill);
               } else {
+                assert(utils.isV3Deposit(historicalDeposit.deposit));
                 const matchedDeposit: V3DepositWithBlock = historicalDeposit.deposit;
```

Comment:
Should be able to drop this one given the assert() above.

```suggestion
                const matchedDeposit = historicalDeposit.deposit;
```